### PR TITLE
[MISC] move llvm's std::span implementation into contrib

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

File: include/seqan3/contrib/span/span [hunk 1]
```diff
@@ -0,0 +1,436 @@
+// -*- C++ -*-
+//!\cond
+//===------------------------------ span ---------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is dual licensed under the MIT and the University of Illinois Open
+// Source Licenses. See LICENSE.TXT for details.
+//
+//===---------------------------------------------------------------------===//
```
Comment (Irallia) [inline include/seqan3/contrib/span/span:10]: I assume that this is intended and still applies?

File: include/seqan3/contrib/span/span [hunk 2]
```diff
@@ -0,0 +1,436 @@
+// -*- C++ -*-
```
Comment (Irallia) [inline include/seqan3/contrib/span/span:1]: Do we need this? What does it mean, just that it is C++ code?

File: include/seqan3/std/span [hunk 3]
```diff
@@ -1,13 +1,10 @@
 // -*- C++ -*-
```
Comment (Irallia) [inline include/seqan3/std/span:1]: And we keep this line?

**Review by Irallia — CHANGES_REQUESTED** [review]
LGFM, just some questions from my side.

File: include/seqan3/contrib/span/span (hunk 1 above)
Comment (marehr) [reply include/seqan3/contrib/span/span:10]: Yep, we copied that span implementation from llvm

File: include/seqan3/contrib/span/span (hunk 2 above)
Comment (marehr) [reply include/seqan3/contrib/span/span:1]: Yes, this is an "editor" standard to set the file extension for extension less files.

File: include/seqan3/std/span (hunk 3 above)
Comment (marehr) [reply include/seqan3/std/span:1]: Yes!

**Review by Irallia — APPROVED** [review]
Okay thanks for the explanation! ❤️

**Comment by marehr** [comment]
Can you @eseiler  take over for @smehringer ?

**Review by eseiler — APPROVED** [review]

**Pull request merged**